etcdmgmt/etcd-mgmt.go

```diff
@@ -160,3 +147,115 @@ func isETCDStartNeeded() (bool, int) {
 	}
 	return start, pid
 }
+
+func EtcdStartInit() (*os.Process, error) {
+	// Check whether etcd environment variable present or not
+	// If it present then start etcd without --initial-cluster flag
+	// other wise start etcd normally.
+
+	file, err := os.Open(etcdConfFile)
+	if err != nil {
+		log.Info("Starting/Restarting etcd for a initial node")
+		return StartInitialEtcd()
+	} else {
+		defer file.Close()
+
+		// Restoring etcd environment variable and starting etcd daemon
+		scanner := bufio.NewScanner(file)
+		scanner.Split(bufio.ScanLines)
+		for scanner.Scan() {
+			linestr := scanner.Text()
+			etcdenv := strings.Split(linestr, "=")
+
+			etcdEnvKey := etcdenv[0]
+			etcdEnvData := etcdenv[1]
+			envlen := len(etcdenv)
+			for i := 2; i < envlen; i++ {
```

Comment:
this loop needed for restoring etcd environment variable during restarting of etcd/glusterd. this loop reading etcd environment variable from the etcdConfFile and spliting   etcd environment key and value. left side of first occurrence of = will be etcdenv key and right side will be value. split function give you array of key/value which are separated by   "=" so at right side we need to join all that key/value, that is done by inner for loop.